00X-tracing.md

```diff
@@ -0,0 +1,581 @@
+   *   invoked only once per event, even for a multi-category event.
+   */
+  on(category: string | string[], listener: Function): void;
+
+  /**
+   * Adds a listener for one or more categories of events.
+   *
+   * @param category Required category name or array of one or more category names to listen to.
+   * @param listener Receives events of the requested category. or categories. A listener is
+   *   invoked only once per event, even for a multi-category event.
+   */
+  addListener(category: string | string[], listener: Function): void;
+
+  /**
+   * Removes a listener for one or more categories of events.
+   *
+   * @param category Required category name or array of one or more category names to stop
+   * listening to.
+   * @param listener The listener to remove.
+   */
+  removeListener(category: string | string[], listener: Function): void;
+
+  /**
+   * Removes all listeners for one or more categories of events, or removes all listeners
+   * for all categories if no category argument is supplied.
+   *
+   * @param category Required category name or array of one or more category names to stop
+   *   listening to.
+   */
+  removeAllListeners(category?: string | string[]): void;
+
+  /**
+   * Gets all the listeners for one or more categories of events.
+   *
+   * @param category Optional category name or array of one or more category names to retrieve
+   *   listeners for.
+   * @returns Array of listeners for the specified categories. Even if a listener is registered
+   *   for more than one of the categories, it is only included in the list once.
+   */
+  listeners(category?: string | string[]): Function[];
+
+  /**
+   * Gets a count of all the listeners for one or more categories of events.
+   *
+   * @param category Optional category name or array of one or more category names to count
+   *   listeners for.
+   * @returns Count of listeners for the specified categories. Even if a listener is registered
+   *   for more than one of the categories, it is only counted once.
+   */
+  listenerCount(category?: string | string[]): number;
+
+  /**
+   * Gets a list of all the categories having registered listeners.
+   */
+  listenerCategories(): string[];
+}
+```
+
+## 4. Console tracing API extensions
+The JavaScript `console` module (in Node.js and the web) already defines a set of basic tracing
+methods that align well with the four tracing event types described above. These methods are NOT
+part of any official standard, though there is
+[a working group that maintains a spec](https://github.com/whatwg/console)
+for the purpose of encouraging interoperability.
+
+The existing Node.js implementation of those Console tracing methods simply logs tracing events to
+the console. This proposal extends those methods with a new optional category parameter to enable
+emitting categorized tracing events in a way that is backward-compatible with existing console APIs.
```

Comment:
I don't like the idea of deviating from the behaviour of the console API spec. If they decide to add their own extra argument in the future, suddenly we have problems.